app/models/miq_worker.rb

```diff
@@ -358,9 +367,23 @@ def containerized_worker?
     self.class.containerized_worker?
   end
 
+  def self.supports_systemd?
+    worker_settings[:systemd_enabled]
+  end
+
+  def self.systemd_worker?
+    MiqEnvironment::Command.supports_systemd? && supports_systemd?
+  end
+
+  def systemd_worker?
+    self.class.systemd_worker?
+  end
+
   def start_runner
     if ENV['MIQ_SPAWN_WORKERS'] || !Process.respond_to?(:fork)
       start_runner_via_spawn
+    elsif systemd_worker?
+      start_systemd_worker
```

Comment:
Do we care that the other two methods here are `runner` and this one is `worker`?

Reply:
So actually at one point I had a method similar to the container one:

```
def start_runner_via_container
  create_container_objects
end

def start_runner_via_systemd
  start_systemd_worker
end
```

It felt pointless just to match the other type's method names, and it doesn't actually start runner directly that's done by run_single_worker right?  I can add it back but it was a one liner method just so the names look nice...

Reply:
Or to be more clear, I'm fine with this the way it is.